flutter-gui-tests/testSrc/io/flutter/tests/gui/ProjectCreator.kt

```diff
@@ -0,0 +1,61 @@
+package io.flutter.tests.gui
+
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.testGuiFramework.framework.Timeouts
+import com.intellij.testGuiFramework.impl.*
+import com.intellij.testGuiFramework.utils.TestUtilsClass
+import com.intellij.testGuiFramework.utils.TestUtilsClassCompanion
+import org.fest.swing.exception.WaitTimedOutError
+
+val GuiTestCase.ProjectCreator by ProjectCreator
+
+// Inspired by CommunityProjectCreator
+class ProjectCreator(guiTestCase: GuiTestCase) : TestUtilsClass(guiTestCase) {
+
+  companion object : TestUtilsClassCompanion<ProjectCreator>({ it -> ProjectCreator(it) })
+
+  private val defaultProjectName = "untitled"
+  private val LOG = Logger.getInstance(this.javaClass)
+
+  fun createProject(projectName: String = defaultProjectName, needToOpenMain: Boolean = true) {
+    with(guiTestCase) {
+      welcomeFrame {
+        actionLink("Create New Project").click()
+        GuiTestUtilKt.waitProgressDialogUntilGone(robot = robot(), progressTitle = "Loading Templates",
+                                                  timeoutToAppear = Timeouts.seconds02)
```

Comment:
~this feels like a race condition waiting to happen.~

**EDIT:** I'm guessing this actually waits for the dialog to appear and then disappear?

Reply:
Exactly, and the timer limits the wait for appearance.